content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -86,46 +86,324 @@ In the above example, this version is `kubelet.config.k8s.io/v1beta1`.
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+clusterDNS:
+  - "192.168.0.2"
+  - "192.168.0.3"
+  - "192.168.0.5"
+```
+
+#### Maps, including Nested Structures
+
+Individual fields in maps, regardless of their value types (boolean, string, etc.), can be selectively overridden.
+However, for `map[string][]string`, the entire list associated with a specific field gets overridden.
+Let's understand this better with an example, particularly on fields like `featureGates` and `staticPodURLHeader`:
+
+Main kubelet configuration file contents:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+featureGates:
+  AllAlpha: false
+  MemoryQoS: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 234APSDFA"
+  - "X-Custom-Header: 123"
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 456"
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+featureGates:
+  MemoryQoS: false
+  KubeletTracing: true
+  DynamicResourceAllocation: true
+staticPodURLHeader:
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 345"
+```
+
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+featureGates:
+  AllAlpha: false
+  MemoryQoS: false
+  KubeletTracing: true
+  DynamicResourceAllocation: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 234APSDFA"
+  - "X-Custom-Header: 123"
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 345"
+```
+
+### Kubelet configuration merging order
+
 On startup, the kubelet merges configuration from:
 
-* Command line arguments (lowest precedence).
-* the kubelet configuration
+* Feature gates specified over the command line (lowest precedence).
+* The kubelet configuration.
 * Drop-in configuration files, according to sort order.
-* Feature gates specified over the command line (highest precedence).
+* Command line arguments exclude feature gates (highest precedence).
+
+{{< note >}}
+The config drop-in dir mechanism for the kubelet is similar but different from how the `kubeadm` tool allows you to patch configuration.
+The `kubeadm` tool uses a specific [patching strategy](/docs/setup/production-environment/tools/kubeadm/control-plane-flags/#patches) for its configuration,
+whereas the only patch strategy for kubelet configuration drop-in files is `replace`. The kubelet determines the order of merges based on sorting the **suffixes** alphanumerically,
+and replaces every field present in a higher priority file.
+{{< /note >}}
 
 This produces the same outcome as if you used the [single configuration file](#create-the-config-file) used in the earlier example.
 
+## Viewing the kubelet configuration
+
```

Comment:
Maybe add motivation for this:  `Since the configuration could now be spread over multiple files with this feature, if someone wants to inspect the final actuated configuration, they can follow these steps: `